3rdParty/iresearch/core/formats/columnstore2.cpp

```diff
@@ -584,108 +623,116 @@ doc_iterator::ptr mask_column::iterator(bool /*consolidation*/) const {
 class dense_fixed_length_column final : public column_base {
  public:
-  static column_ptr read(
-    std::optional<std::string>&& name,
-    bstring&& payload,
-    column_header&& hdr,
-    column_index&& index,
-    index_input& index_in,
-    const index_input& data_in,
-    compression::decompressor::ptr&& inflater,
-    encryption::stream* cipher);
-
-  dense_fixed_length_column(
-      std::optional<std::string>&& name,
-      bstring&& payload,
-      column_header&& hdr,
-      column_index&& index,
-      const index_input& data_in,
-      compression::decompressor::ptr&& inflater,
-      encryption::stream* cipher,
-      uint64_t data,
-      uint64_t len)
-    : column_base{ std::move(name), std::move(payload),
-                   std::move(hdr), std::move(index),
-                   data_in, cipher },
-      inflater_{std::move(inflater)},
-      data_{data},
-      len_{len} {
+  template<typename ColumnType>
+  static column_ptr read(std::optional<std::string>&& name, bstring&& payload,
+                         column_header&& hdr, column_index&& index,
+                         index_input& index_in, const index_input& data_in,
+                         compression::decompressor::ptr&& inflater,
+                         encryption::stream* cipher) {
+    const uint64_t len = index_in.read_long();
+    const uint64_t data = index_in.read_long();
+
+    return memory::make_unique<ColumnType>(
+        std::move(name), std::move(payload), std::move(hdr), std::move(index),
+        data_in, std::move(inflater), cipher, data, len);
+  }
+
+  dense_fixed_length_column(std::optional<std::string>&& name,
+                            bstring&& payload, column_header&& hdr,
+                            column_index&& index, const index_input& data_in,
+                            compression::decompressor::ptr&& inflater,
+                            encryption::stream* cipher, uint64_t data,
+                            uint64_t len)
+      : column_base{std::move(name), std::move(payload),
+                    std::move(hdr),  std::move(index),
+                    data_in,         cipher},
+        inflater_{std::move(inflater)},
+        data_{data},
+        len_{len} {
     assert(header().docs_count);
     assert(ColumnType::kDenseFixed == header().type);
   }
 
   virtual doc_iterator::ptr iterator(bool /*consolidation*/) const override;
 
+  void make_buffered(
+      irs::index_input& in, const memory_accounting_f& memory_accounter,
+      range<std::unique_ptr<column_reader>> next_sorted_columns) override {
+    buffered_memory_accounter_ = memory_accounter;
+    auto& hdr = mutable_header();
+    auto const data_size = len_ * hdr.docs_count;
+    auto const bitmap_size =
+        calculate_bitmap_size(in.length(), next_sorted_columns);
+    auto const total_size = data_size + bitmap_size;
+    if (!allocate_buffered_memory(total_size)) {
+      return;
```

Comment:
If we return here, we will have already modified `buffered_memory_accounter_`, even though we are not buffered.

Reply:
Yeah, not a good semantics. Moved that to column_base private